source/h1_encoder.c

```diff
@@ -610,15 +628,16 @@ static int s_state_fn_unchunked_body(struct aws_h1_encoder *encoder, struct aws_
  * Encoder is essentially "paused" here if no chunks are available. */
 static int s_state_fn_chunk_next(struct aws_h1_encoder *encoder, struct aws_byte_buf *dst) {
     (void)dst;
+    AWS_ASSERT(encoder->message->chunk_list); /* we let this be NULL in tests, but not in real world */
 
-    if (aws_linked_list_empty(&encoder->message->chunk_list)) {
+    if (aws_linked_list_empty(encoder->message->chunk_list)) {
```

Comment:
Could this just check null on ```chunk_list``` and no assert needed? Or is that a THIS SHOULD NEVER EVER BE situation? Why would we ever allow it to be null in tests if that's the case?

Reply:
I'll fix the tests to not be lazy jerks